Common/Securities/CashAccountMarginModel.cs

```diff
@@ -0,0 +1,242 @@
+                        // we have this much 'cash' that can be sold
+                        var feesInBaseCurrency = _cashBook.Convert(fees, CashBook.AccountCurrency, baseCurrency.BaseCurrencySymbol);
+                        return order.AbsoluteQuantity + feesInBaseCurrency;
+                    }
+
+                    // assuming these are 'units' of stock priced in the account currency
+                    return marginRequiredInAccountCurrency;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        /// <summary>
+        /// Gets the margin cash available for a trade in units of the currency required.
+        /// When increasing a position, this is the quote currency, when decreasing a position this is the base currency
+        /// </summary>
+        /// <param name="portfolio">The algorithm's portfolio</param>
+        /// <param name="security">The security to be traded</param>
+        /// <param name="direction">The direction of the trade</param>
+        /// <returns>The margin available for the trade</returns>
+        public decimal GetMarginRemaining(SecurityPortfolioManager portfolio, Security security, OrderDirection direction)
+        {
+            var baseCurrency = security as IBaseCurrencySymbol;
+            var targetCurrency = string.Empty;
+            var openOrdersTotalValue = 0m;
+            var openOrdersTotalQuantity = 0m;
+            var symbolDirectionPairs = new Dictionary<Symbol, OrderDirection>();
+
+            if (baseCurrency != null)
+            {
+                // find the target currency for the requested direction and the securities potentially involved
+                targetCurrency = direction == OrderDirection.Buy
+                    ? security.QuoteCurrency.Symbol
+                    : baseCurrency.BaseCurrencySymbol;
+
+                foreach (var portfolioSecurity in portfolio.Securities.Values)
+                {
+                    var basePortfolioSecurity = portfolioSecurity as IBaseCurrencySymbol;
+                    if (basePortfolioSecurity == null) continue;
+
+                    if (basePortfolioSecurity.BaseCurrencySymbol == targetCurrency)
+                    {
+                        symbolDirectionPairs.Add(portfolioSecurity.Symbol, OrderDirection.Sell);
+                    }
+                    else if (portfolioSecurity.QuoteCurrency.Symbol == targetCurrency)
+                    {
+                        symbolDirectionPairs.Add(portfolioSecurity.Symbol, OrderDirection.Buy);
+                    }
+                }
+            }
+
+            // fetch open orders with matching symbol/side
+            var openOrders = portfolio.Transactions.GetOpenOrders(x =>
+                {
+                    OrderDirection dir;
+                    return symbolDirectionPairs.Count == 0 ||
+                           symbolDirectionPairs.TryGetValue(x.Symbol, out dir) && dir == x.Direction;
```

Comment:
Awesome! Out of curiosity, what is the case where `symbolDirectionPairs.Count == 0` ? I'm guessing this is the equity/stock case? I think even in that case we want to have symbol direction pairs for all the stocks' quote currency?